### PR TITLE
[flutter_tool] Re-work analytics events to use labels and values

### PR DESCRIPTION
**Pull request by zanderso**
## Description

This PR cleans up analytics events by making use of labels and values that can be attached to events instead of relying only on the category, action, and custom dimensions.

## Tests

I added the following tests:

Adjusted existing tests to match the new event layout.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
The value was a pre-existing parameter that we weren't using? Does this replace custom dimensions for events with only one meaningful numeric value?

**Comment by zanderso** [comment]
Yes, we weren't using the pre-existing value parameter.

As for whether it can always replace a custom dimension, I think it depends. Since we can't give the value a separate name, the value would have to obviously be the value for the event. Otherwise a custom dimension is probably better.

File: packages/flutter_tools/lib/src/reporting/disabled_usage.dart [hunk 1]
```diff
@@ -27,7 +27,13 @@ class DisabledUsage implements Usage {
   void sendCommand(String command, { Map<String, String> parameters }) { }
 
   @override
-  void sendEvent(String category, String parameter, { String label, Map<String, String> parameters }) { }
+  void sendEvent(
+    String category,
+    String parameter, {
+    String label,
+    int value,
+    Map<String, String> parameters
```
Comment (blasten) [inline packages/flutter_tools/lib/src/reporting/disabled_usage.dart]: trivial nit: trailing comma

Comment (zanderso) [reply packages/flutter_tools/lib/src/reporting/disabled_usage.dart]: Thanks! Fixed.

**Review by blasten — APPROVED** [review]
Thanks!

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**